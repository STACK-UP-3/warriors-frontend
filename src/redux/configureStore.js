import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducersCombination from './reducers/index';



export default ()=>{
    const store = createStore(
        reducersCombination, {}, 
        composeWithDevTools(applyMiddleware(thunk,promise))
    );
    
    return store;
}
