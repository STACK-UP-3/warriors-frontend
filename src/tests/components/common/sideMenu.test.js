import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Sidemenu from '../../../components/common/sideMenu';
import configureStore from '../../../redux/configureStore';
import { MemoryRouter } from 'react-router-dom';

const store = configureStore();

describe('*************** Testing the SideMenu component ***************',()=>{
    let wrapper;
    const { location } = window;

    beforeEach(()=>{
        delete window.location;
		window.location = { pathname:'/dashboard' };
        
        wrapper = mount( 
            <Provider store={ store }> 
                <MemoryRouter initialEntries={['/dashboard']}>
                    <Sidemenu />
                </MemoryRouter>
            </Provider>
        );
        
    });

    afterAll(() => {
		window.location = location;
    });

    it('Should render the dashboard component correctly', ()=>{
        expect(wrapper).toMatchSnapshot();
    });
    
    it('Should render the Trip request component correctly', ()=>{
        window.location = { pathname:'/trips/oneway' };
        
        const wrapper = mount( 
            <Provider store={ store }> 
                <MemoryRouter initialEntries={['/trips/oneway']}>
                    <Sidemenu />
                </MemoryRouter>
            </Provider>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});
