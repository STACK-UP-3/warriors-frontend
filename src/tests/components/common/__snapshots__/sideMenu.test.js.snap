// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*************** Testing the SideMenu component *************** Should render the Trip request component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/trips/oneway",
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "po2gnj",
              "pathname": "/trips/oneway",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "po2gnj",
            "pathname": "/trips/oneway",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <_default>
        <div
          className="sideMenuContainer"
        >
          <div
            className="profile"
          >
            <img
              src={
                Object {
                  "process": [Function],
                }
              }
            />
             
            <a
              className="btn-floating btn-small halfway-fab waves-effect waves-light blue lighten-5"
            >
              <i
                className="large material-icons indigo-text darken-4"
              >
                mode_edit
              </i>
            </a>
          </div>
          <div
            className="menu"
          >
            <Link
              className="menuItem black-text sidenav-close"
              to="/dashboard"
            >
              <LinkAnchor
                className="menuItem black-text sidenav-close"
                href="/dashboard"
                navigate={[Function]}
              >
                <a
                  className="menuItem black-text sidenav-close"
                  href="/dashboard"
                  onClick={[Function]}
                >
                  <h6>
                    Dashboard
                  </h6>
                  <i
                    className="material-icons"
                  >
                    pie_chart
                  </i>
                </a>
              </LinkAnchor>
            </Link>
            <Link
              className="activeMenuItem black-text sidenav-close"
              to="/trips/oneway"
            >
              <LinkAnchor
                className="activeMenuItem black-text sidenav-close"
                href="/trips/oneway"
                navigate={[Function]}
              >
                <a
                  className="activeMenuItem black-text sidenav-close"
                  href="/trips/oneway"
                  onClick={[Function]}
                >
                  <h6>
                    Make trip request
                  </h6>
                  <i
                    className="material-icons"
                  >
                    airplanemode_active
                  </i>
                </a>
              </LinkAnchor>
            </Link>
            <Link
              className="menuItem black-text sidenav-close"
              to="/viewtrips/accepted"
            >
              <LinkAnchor
                className="menuItem black-text sidenav-close"
                href="/viewtrips/accepted"
                navigate={[Function]}
              >
                <a
                  className="menuItem black-text sidenav-close"
                  href="/viewtrips/accepted"
                  onClick={[Function]}
                >
                  <h6>
                    My Trips
                  </h6>
                  <i
                    className="material-icons"
                  >
                    card_travel
                  </i>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </_default>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`*************** Testing the SideMenu component *************** Should render the dashboard component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(Symbol.observable): [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/dashboard",
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "6tsk7b",
              "pathname": "/dashboard",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "6tsk7b",
            "pathname": "/dashboard",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <_default>
        <div
          className="sideMenuContainer"
        >
          <div
            className="profile"
          >
            <img
              src={
                Object {
                  "process": [Function],
                }
              }
            />
             
            <a
              className="btn-floating btn-small halfway-fab waves-effect waves-light blue lighten-5"
            >
              <i
                className="large material-icons indigo-text darken-4"
              >
                mode_edit
              </i>
            </a>
          </div>
          <div
            className="menu"
          >
            <Link
              className="activeMenuItem black-text sidenav-close"
              to="/dashboard"
            >
              <LinkAnchor
                className="activeMenuItem black-text sidenav-close"
                href="/dashboard"
                navigate={[Function]}
              >
                <a
                  className="activeMenuItem black-text sidenav-close"
                  href="/dashboard"
                  onClick={[Function]}
                >
                  <h6>
                    Dashboard
                  </h6>
                  <i
                    className="material-icons"
                  >
                    pie_chart
                  </i>
                </a>
              </LinkAnchor>
            </Link>
            <Link
              className="menuItem black-text sidenav-close"
              to="/trips/oneway"
            >
              <LinkAnchor
                className="menuItem black-text sidenav-close"
                href="/trips/oneway"
                navigate={[Function]}
              >
                <a
                  className="menuItem black-text sidenav-close"
                  href="/trips/oneway"
                  onClick={[Function]}
                >
                  <h6>
                    Make trip request
                  </h6>
                  <i
                    className="material-icons"
                  >
                    airplanemode_active
                  </i>
                </a>
              </LinkAnchor>
            </Link>
            <Link
              className="menuItem black-text sidenav-close"
              to="/viewtrips/accepted"
            >
              <LinkAnchor
                className="menuItem black-text sidenav-close"
                href="/viewtrips/accepted"
                navigate={[Function]}
              >
                <a
                  className="menuItem black-text sidenav-close"
                  href="/viewtrips/accepted"
                  onClick={[Function]}
                >
                  <h6>
                    My Trips
                  </h6>
                  <i
                    className="material-icons"
                  >
                    card_travel
                  </i>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </_default>
    </Router>
  </MemoryRouter>
</Provider>
`;
